lib/palindrome_check.rb

```diff
@@ -1,7 +1,25 @@
-# A method to check if the input string is a palindrome.
-# Return true if the string is a palindrome. Return false otherwise.
-# Time complexity: ?
-# Space complexity: ?
+# A method to check if the input my_phrase is a palindrome.
+# Return true if the my_phrase is a palindrome. Return false otherwise.
+# Time complexity:  O(n), n is the number of characters or spaces in my_phrase
+# Space complexity: O(1)
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  if my_phrase == nil
+    return false
+  end
+  i = 0
+  j = my_phrase.length - 1
+  while i < j
+    while i < my_phrase.length && my_phrase[i] == " "
+      i += 1
+    end
+    while j >= 0 && my_phrase[j] == " "
+      j -= 1
+    end
+    if i < my_phrase.length && j >= 0 && my_phrase[i] != my_phrase[j]
```

Comment:
It shouldn't be possible for `i` to exceed the length of the string or for `j` to be less than 0. I think you can safely remove this part:

```i < my_phrase.length && j >= 0```